### ISSUE TITLE
Initial database setup appear incorrect on Unraid

### ISSUE THREAD

**Issue by mikewagnercmp** [jokob-sk/Pi.Alert#491]
## Describe the issue

> When submitting an issue ❗[enable debug](https://github.com/jokob-sk/Pi.Alert/blob/main/docs/DEBUG_TIPS.md)❗ and [have a look at the docs](https://github.com/jokob-sk/Pi.Alert/tree/main/docs)

Installed a fresh PiAlert install from the CA template on my Unraid machine  - running 6.11.5 of unraid

When the container attempts to start, the following is  logged

```
11/01/2023
07:57:54 PM
Nothing to do here.
11/01/2023
07:57:54 PM
Starting nginx: nginx.
11/01/2023
07:57:54 PM
Traceback (most recent call last):
11/01/2023
07:57:54 PM
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
11/01/2023
07:57:54 PM
    return _run_code(code, main_globals, None,
11/01/2023
07:57:54 PM
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
11/01/2023
07:57:54 PM
    exec(code, run_globals)
11/01/2023
07:57:54 PM
  File "/home/pi/pialert/pialert/__main__.py", line 179, in <module>
11/01/2023
07:57:54 PM
    sys.exit(main())       
11/01/2023
07:57:54 PM
  File "/home/pi/pialert/pialert/__main__.py", line 75, in main
11/01/2023
07:57:54 PM
    db.upgradeDB()
11/01/2023
07:57:54 PM
  File "/home/pi/pialert/pialert/database.py", line 228, in upgradeDB
11/01/2023
07:57:54 PM
    self.sql.execute("""INSERT INTO Plugins_Objects (
11/01/2023
07:57:54 PM
sqlite3.OperationalError: no such table: Plugins_Objects
11/01/2023
07:57:54 PM
19:57:54 [MAIN] Setting up ...
11/01/2023
07:57:54 PM
19:57:54 [conf.tz] Setting up ...
11/01/2023
07:57:54 PM
19:57:54 
11/01/2023
07:57:54 PM
11/01/2023
07:57:54 PM
19:57:54 The container restarted (started). If this is unexpected check https://bit.ly/PiAlertDebug for troubleshooting tips.
11/01/2023
07:57:54 PM
19:57:54 
11/01/2023
07:57:54 PM
11/01/2023
07:57:54 PM
19:57:54 Permissions check (All should be True)
11/01/2023
07:57:54 PM
19:57:54 ------------------------------------------------
11/01/2023
07:57:54 PM
19:57:54   /config/pialert.conf |  READ  | True
11/01/2023
07:57:54 PM
19:57:54   /config/pialert.conf |  WRITE | True
11/01/2023
07:57:54 PM
19:57:54   /db/pialert.db       |  READ  | True
11/01/2023
07:57:54 PM
19:57:54   /db/pialert.db       |  WRITE | True
11/01/2023
07:57:54 PM
19:57:54 ------------------------------------------------
11/01/2023
07:57:54 PM
19:57:54 [Setup] Attempting to fix permissions.
11/01/2023
07:57:54 PM
19:57:54 [Setup] Attempting to fix permissions.
11/01/2023
07:57:54 PM
19:57:54 [Database] Opening DB
11/01/2023
07:57:54 PM
19:57:54 [upgradeDB] Re-creating Settings table
11/01/2023
07:57:54 PM
19:57:54 [upgradeDB] Re-creating Pholus_Scan table
11/01/2023
07:57:54 PM
19:57:54 [upgradeDB] Re-creating Parameters table
```

## Paste your `pialert.conf` (remove personal info)

Unable to get the container to start to grab the configs without doing extra messing around.

**Comment by jokob-sk** [comment]
Hey there,

I don't have Unraid running myself so I'll need a bit more info from your end.

1. What's a CA template?
2. Do you use a standard Docker environment?
3. What is your docker-compose file?

It seems like eventually, the container starts, or is it stuck in a restart loop?

It might be permissions related or the DB file might be missing/not mapped correctly?

Thansk for more info!
j

**Comment by mikewagnercmp** [comment]
So in unraid, a CA template is just their wrapper for a docker container that you grab from their app center. The app looked like it was official so I assumed you would know about it, it did have your user I’d on the app in tje app center but anyone could have done that.

when I am in front of my computer again I’ll try to grab the docker command that runs. I’ll also post the docker image tje app is grabbing from docker hub.

Unraid makes it easy to install everything,  it by making it simple, you have to monkey with stuff when it doesn’t work like you expect.

**Comment by mikewagnercmp** [comment]
Here is the docker run - 

```
docker run
  -d
  --name='PiAlert'
  --net='host'
  -e TZ="America/New_York"
  -e HOST_OS="Unraid"
  -e HOST_HOSTNAME="UnRaidNas"
  -e HOST_CONTAINERNAME="PiAlert"
  -e 'TZ'='America/New_York'
  -e 'TCP_PORT_20211'='20211'
  -l net.unraid.docker.managed=dockerman
  -l net.unraid.docker.webui='http://[IP]:[PORT:20211]'
  -l net.unraid.docker.icon='https://raw.githubusercontent.com/pucherot/Pi.Alert/main/front/img/pialertLogoGray50.png'
  -v '/mnt/user/appdata/pialert/config':'/home/pi/pialert/config':'rw'
  -v '/mnt/user/appdata/pialert/db':'/home/pi/pialert/db':'rw'
  -v '/mnt/user/appdata/pihole-dot-doh/pihole/pihole-FTL.db':'/etc/pihole/pihole-FTL.db':'rw' 'jokobsk/pi.alert'
fab648cfe2870b62ae9909b2d4f8521c9ba8c32c00a37bfe8055ee246bd559a6
```


Its pulling from dockerhub https://registry.hub.docker.com/r/jokobsk/pi.alert

the person who created the template is not you however, so if you are not able to assist I understand, the person left the support forum on their template for here lol.

I can try rolling my own as well .

Mike

**Comment by mikewagnercmp** [comment]
interestingly, the pialert_dev image works for me, but the normal latest does not

**Comment by jokob-sk** [comment]
@mikewagnercmp Thanks for the details and testing the _dev mage - that means that hopefully I won't break anything and the next release this template will start working again 👍

I'll mark this as solved in the dev image for now

**Comment by jokob-sk** [comment]
Released in 23.11.11
